src/Jackalope/Property.php

```diff
@@ -709,25 +685,44 @@ public function _setValue($value, $type = PropertyType::UNDEFINED)
          */
 
         if (PropertyType::UNDEFINED === $type) {
-            $type = $this->valueConverter->determineType(is_array($value) ? reset($value) : $value);
+            if (is_array($value) && 0 === count($value)) {
+                $type = PropertyType::STRING;
```

Comment:
the problem is that when its an array, we `reset` which is an error on an empty array. we could make the value converter accept arrays as well and let it do the logic. will do that.